src/Sarif/Visitors/GitHubDspIngestionVisitor.cs

```diff
@@ -78,15 +161,11 @@ public override Result VisitResult(Result node)
             {
                 foreach (Location relatedLocation in node.RelatedLocations)
                 {
-                    // DSP requires that every related location have a message. It's
-                    // not clear why this requirement exists, as this data is mostly 
-                    // used to build embedded links from results (where the link
-                    // anchor text actually resides).
                     if (string.IsNullOrEmpty(relatedLocation.Message?.Text))
                     {
                         relatedLocation.Message = new Message
                         {
-                            Text = "[No message provided.]" 
+                            Text = PlaceholderRelatedLocationMessage
```

Comment:
>Text [](start = 28, length = 4)

I thought that Messages could have either a Text or Id property to define the string. Here, if the message contains an Id, we're adding a text string. Is that OK?

Reply:
... I should have added, I realize the previous code did this too, but I thought I'd ask anyway.

---
In reply to: [483750179](https://github.com/microsoft/sarif-sdk/pull/2061#discussion_r483750179) [](ancestors = 483750179)

Reply:
It's a good question. We're compensating here for a deficiency in the DSP -- the DSP looks for the `Text` property.

---
In reply to: [483750179](https://github.com/microsoft/sarif-sdk/pull/2061#discussion_r483750179) [](ancestors = 483750179)

Reply:
... when in fact it shouldn't care about the presence of related location messages at all.

---
In reply to: [483751157](https://github.com/microsoft/sarif-sdk/pull/2061#discussion_r483751157) [](ancestors = 483751157,483750179)